### ISSUE TITLE
implement "b3 single" header format

### ISSUE THREAD

**Issue by adriancole** [Financial-Times/tapper#13]
As discussed on https://github.com/openzipkin/b3-propagation/issues/21 and first implemented here: https://github.com/openzipkin/brave/blob/master/brave/src/main/java/brave/propagation/B3SingleFormat.java https://github.com/openzipkin/brave/blob/master/brave/src/test/java/brave/propagation/B3SingleFormatTest.java

Let's support at least reading "b3" header from a single string, most commonly traceid-spanid-1
It would also be nice to support optionally writing this, especially in message providers or others with constrained environments.

Brave currently has a property like this, but its name could change with feedback:
```java
    /**
     * When true, only writes a single {@link B3SingleFormat b3 header} for outbound propagation.
     *
     * <p>Use this to reduce overhead. Note: normal {@link Tracing#propagation()} is used to parse
     * incoming headers. The implementation must be able to read "b3" headers.
     */
    public Builder b3SingleFormat(boolean b3SingleFormat) {
      this.b3SingleFormat = b3SingleFormat;
      return this;
}
```

**Comment by ellispritchard** [comment]
Implemented in [tapper_plug 0.5.0](https://github.com/Financial-Times/tapper_plug/commit/8a73017d59e827e5a4ecb7b84eb529e8fced44af).

Decodes incoming headers in either original b3 or b3 single format, adds support for encoding to b3 single for downstream propagation.

NB support for propagating only a sampling decision to Tapper has never been supported, headers without trace and span id are ignored. Will support this as part of a coming refactor around trace id representation.

**Comment by adriancole** [comment]
thanks for the help, Ellis! Many folks didn't know about the sampled only
thing, so I moved a diagram to the top here
https://github.com/openzipkin/b3-propagation/pull/32
